### PR TITLE
fix(e12): unresolved-getter error semantics + parseInclude strictness (must-fix bundle for v1.4.0)

### PR DESCRIPTION
**Pull request by y1o1**
## Summary

E12 follow-on must-fix bundle for v1.4.0 release. Three issues triaged via cross-repo audit; ships with TDD-style regression tests and multi-agent review (Claude PASS, Codex PASS).

- **#114 (already-fixed, regression-pinned)**: `Config.resolveWith(source, { allowUnresolved: true })` placeholder tracking. Verified addressed in PR #117 squash; added pinning tests as regression guard.
- **#116 (real fix)**: Unresolved-Config getters (`getConfig` / `getList` / `requireScalar`) previously threw `NotResolvedError` for any missing path. Now gated on `_resObjRootSubtreeHasPlaceholders(path)`; truly-missing paths fall through to `ConfigError("path not found")` as on resolved configs.
- **#113 (real fix)**: Cross-impl gap from [go.hocon#101](https://github.com/o3co/go.hocon/pull/101) (v1.3.1). Two convergent `parseInclude` bugs:
  - **Issue 2 (false-match)**: `innerPrefix.startsWith('file')` matched `fileX(`, same shape for `urlencode(`, `classpathish(`, `packagex(`. Tightened to exact match or `startsWith('X(')`. After all qualifier branches, any non-empty `innerPrefix` raises a parse error rather than falling through to bare-path semantics.
  - **Issue 1 (silent swallow)**: `parseQuotedPathSkipWrapper` advanced unconditionally past `,`, `=`, identifiers until finding a quoted string. Inputs like `include file () b = "x"` silently bound the path to `"x"` and dropped `b = `. Restricted via new `isIncludeWrapperToken` whitelist.
  - Issue 3 (whitespace nested form `include required ( file("foo"))`) deferred per the issue body's "minor — separate fix" note.

## Behavior change

`parseInclude` now rejects inputs that previously silently mis-parsed and produced a `ResolveError` at file-not-found time. Per project policy (1.x library implementing the external HOCON spec): spec-compliance fixes are bug fixes, not breaking changes — even when observable behavior changes. The pre-fix behavior was silently routing the wrong qualifier and producing a misleading error; the post-fix behavior raises an immediate `ParseError` at the actual location of the problem.

## Test plan

- [x] `tests/ts114-resolve-with-allow-unresolved.test.ts` — 3 cases (already-resolved-by-source / still-unresolved-after-source / fully-satisfied-after-source)
- [x] `tests/ts116-getter-missing-path.test.ts` — 3 cases (truly-missing path → ConfigError / placeholder path → NotResolvedError / resolved path → value)
- [x] `tests/ts113-parse-include-strict.test.ts` — 10 cases (silent-swallow reproductions, false-match reproductions, valid forms, edge cases)
- [x] Full suite: 984 passed / 9 expected-fail / 5 skipped (developed from 981 pre-fix, +3 new test files)
- [x] `npx tsc --noEmit` clean
- [x] Multi-agent review (Claude + Codex) PASS, no must-fix items

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/internal/parser/parser.ts [hunk 1]
```diff
@@ -302,17 +315,45 @@ class Parser {
...
+      throw new ParseError(`unexpected token after include path: ${desc}`, after.line, after.col)
+    }
     return path
   }
 
+  private isIncludeWrapperToken(tok: Token): boolean {
+    if (tok.kind !== 'unquoted') return false
+    const v = tok.value
+    if (v === '(' || v === ')') return true
+    if (v === 'file' || v === 'file(') return true
```
[the remark this answers is not in the document]
Comment (y1o1) [reply src/internal/parser/parser.ts]: Fixed in 6cab1c2 — narrowed `isIncludeWrapperToken` to bare `(` only. `)` is no longer in the pre-path allowlist; closing-paren before the quoted string now triggers `ParseError` (e.g. `include required( ) "x"`, `include file ) "x"`). Closing parens are still recognised in the post-path loop only. Regression tests added in tests/ts113-parse-include-strict.test.ts (`rejects required( ) "x"`, `rejects file ) "x"`).

File: src/internal/parser/parser.ts [hunk 2]
```diff
@@ -302,17 +315,45 @@ class Parser {
...
+    if (v === 'package' || v === 'package(') return true
+    if (v === 'url' || v === 'url(') return true
+    if (v === 'classpath' || v === 'classpath(') return true
```
[the remark this answers is not in the document]
Comment (y1o1) [reply src/internal/parser/parser.ts]: Fixed in 6cab1c2 — narrowed `isIncludeWrapperToken` to bare `(` only. Qualifier keywords (`file`/`package`/`url`/`classpath`, with or without paren) are no longer skip-able pre-path. Whitespace-nested unsupported forms like `include required ( url ("a.conf"))` and `include required ( classpath ("a.conf"))` now raise `ParseError` instead of silently routing to bare semantics. The `file` qualifier in the bare-required wrapper is consumed at the call site with an explicit `this.advance()` so the helper does not need to allow keywords. Issue 3 (proper recognition of `include required ( file("foo"))`) remains tracked as a follow-up — those inputs now raise a loud `ParseError` rather than silently mis-routing. Regression tests added covering url/classpath cases plus valid-form anchors.
